File: docker/runner.py
import sys
import json
import time
from io import StringIO
from contextlib import redirect_stdout, redirect_stderr

def execute_code(code_str):
    start_time = time.time()
    stdout = StringIO()
    stderr = StringIO()
    
    try:
        with redirect_stdout(stdout), redirect_stderr(stderr):
            exec(code_str)
        status = "success"
    except Exception as e:
        print(str(e), file=sys.stderr)
        status = "failure"
    
    execution_time = f"{time.time() - start_time:.3f}s"
    
    return {
        "stdout": stdout.getvalue(),
        "stderr": stderr.getvalue(),
        "executionTime": execution_time,
        "status": status
    }

if __name__ == "__main__":
    # Read code from stdin
    code = sys.stdin.read()
    result = execute_code(code)
    print(json.dumps(result))

File: docker/runner.js
const vm = require('vm');

async function executeCode(code) {
    const startTime = process.hrtime();
    const context = {};
    
    try {
        // Create a new context for code execution
        vm.createContext(context);
        
        // Execute the code
        vm.runInContext(code, context);
        
        const [seconds, nanoseconds] = process.hrtime(startTime);
        const executionTime = `${seconds}.${Math.floor(nanoseconds / 1000000)}s`;
        
        return {
            stdout: '',  // Captured through console.log redirection
            stderr: '',
            executionTime,
            status: 'success'
        };
    } catch (error) {
        const [seconds, nanoseconds] = process.hrtime(startTime);
        const executionTime = `${seconds}.${Math.floor(nanoseconds / 1000000)}s`;
        
        return {
            stdout: '',
            stderr: error.message,
            executionTime,
            status: 'failure'
        };
    }
}

// Read input from stdin
let code = '';
process.stdin.on('data', chunk => {
    code += chunk;
});

process.stdin.on('end', async () => {
    const result = await executeCode(code);
    console.log(JSON.stringify(result));
});
